hw3/hw3-makinunal/hw3-akinunal.js

```diff
@@ -0,0 +1,131 @@
+/* 1. write a constructor function PersonFunc which accepts the person data like in file people.data.js and implements next methods:
+getFullName() => returns string in format "[FirstName] [LastName]", e.g. 'Viktoriia Chernyshuk';
+getAge() => returns age of person in format "[number] years", e.g. "25 years";
+getFullAddress() => returns address string in format "[country], [city], [street], [house]/[apartment]". Field apartment is optional. E.g. "Ukraine, Lviv, Doroshenka street, 5/8" and "Ukraine, Lviv, Muchna street, 7".
+Please implement error handling in case if some fields are not available.
+*/
+import { SarahMay, JeremieBrown, Elizabeth } from './people.data.js';
+
+console.log('------ QUESTION: 1 ------')
+
+function PersonFunc(prs) {
+    this.person = prs
+    this.fullName = function () {
+        if (!this.person.firstName || !this.person.lastName) {
+            return `Unknown first/last name.`
+        }
+        return `${this.person.firstName} ${this.person.lastName}`;
+    }
+
+    this.getAge = function () {
+        let date = new Date()
+        let currentYear = date.getFullYear();
+        if (!this.person.birthDate || this.person.birthDate.length != 10) {
+            return `Unknown birth date.`
+        }
+        let yearOfBirth = parseInt(this.person.birthDate.slice(6, 10));
+        return `${currentYear - yearOfBirth} years.`
+    }
+
+    this.getFullAddress = function () {
+        const { country, city, street, apartment, house } = this.person.address;
+        if (!country || !city || !street || !house) {
+            return `Please fill all the fields.`
+        }
+        let result = `${country}, ${city}, ${street}, ${house}`
+        if (apartment) {
+            result = `${result} / ${apartment}`
+        }
+        return result
+    }
+}
+
+let sarahCons = new PersonFunc(SarahMay);
+let jeremieCons = new PersonFunc(JeremieBrown);
+let elizabethCons = new PersonFunc(Elizabeth);
+
+console.log(sarahCons.fullName());
+console.log(jeremieCons.getAge());
+console.log(elizabethCons.getFullAddress());
+
+/* 2.write a constructor function WorkingPersonFunc which inherits from PersonFunc and implements additional method:
+getProfessionalNameAndRank() => returns string in format "[FirstName] [LastName], [job title], job experience [job experience duration] years", e.g. "Viktoriia Chernyshuk, Front-end developer, job experience 2.5 years".
+Please implement error handling in case if some fields are not available.
+*/
+console.log('------ QUESTION: 2 ------')
+
+function WorkingPersonFunc(person) {
+    PersonFunc.call(this, person);
+    this.getProfessionalNameAndRank = function () {
+        return `${this.person.firstName} ${this.person.lastName}, ${this.person.job.title}, job experience ${this.person.job.experience} years.`
```

Comment:
for getting `{this.person.firstName} ${this.person.lastName}` you could use parent's getFullName() method if you'd move it to prototype.
Also would be nice to check whether job title and experience are present so we did not get undefined in resulting string